bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1769,7 +1769,7 @@ void handleUnrecoverableErrorDuringAdd(int rc) {
         if (getLedgerMetadata().getState() == LedgerMetadata.State.IN_RECOVERY) {
             // we should not close ledger if ledger is recovery mode
             // otherwise we may lose entry.
-            errorOutPendingAdds(rc);
+            executeOrdered(() -> errorOutPendingAdds(rc));
```

Comment:
@horizonzy  I may not have explained it clearly. Although this is changed to an async operation, this code block will never be executed, because when in IN_RECOVERY mode, the method of ReadOnlyLedgerHandle must be called.
What do you think if I roll back the changes to LedgerHandle and only keep the changes to ReadOnlyLedgerHandle

Reply:
I see.

Reply:
I see.